### ISSUE TITLE
[hyprexpo] Keyboard redirection

### ISSUE THREAD

**Issue by Prabhuomkar9** [hyprwm/hyprland-plugins#195]
When expo is open
1. keyboard inputs are currently mapped to the current workspace, which is not a good experience considering the workspace size just goes zoosh(small) => one might type something without realising
2. what if we could instead block most keys and only allow keys like
- `Left` `Right` `Up` `Down` for navigation
- `Space` `Enter` to select
- `Esc` potentially to close
3. potential future being we can go outside the limit 9 (columns=3) and as we navigate to maybe the 8th one the entire expo move one step left and right for start(given not workspace 1)

**Comment by nnra6864** [comment]
I absolutely agree with this.
I always wanted it to support numeric input, for example, pressing 3 would switch you to the 3rd workspace etc.

**Comment by Prabhuomkar9** [comment]
> I absolutely agree with this.
> I always wanted it to support numeric input, for example, pressing 3 would switch you to the 3rd workspace etc.

Pressing 3 will move to workspace 3 can be achieved by binds right

something like 
`bind = Super, 3, workspace, 3`
or do u mean like inside hyprexpo we navigate with numbers?

**Comment by nnra6864** [comment]
I meant inside hyprexpo.
What you said does indeed work, but say you are already on the 3rd workspace.
If you do super+3, nothing will happen because you are already on the 3rd workspace.

**Comment by Fossu** [comment]
Agreed with a need of keyboard navigation and previous comments, It would be nice to be able to have a new set of binds inside hyprexpo, like press 3 to go to 3rd workspace, press j to move down in the grid etc.
